system/CodeIgniter.php

```diff
@@ -528,7 +528,7 @@ protected function detectEnvironment()
     {
         // Make sure ENVIRONMENT isn't already set by other means.
         if (! defined('ENVIRONMENT')) {
-            define('ENVIRONMENT', $_SERVER['CI_ENVIRONMENT'] ?? 'production');
+            define('ENVIRONMENT', getenv('CI_ENVIRONMENT') ?? 'production');
```

Comment:
```suggestion
            define('ENVIRONMENT', getenv('CI_ENVIRONMENT') ?: 'production');
```

ref:

-  https://phpstan.org/r/f3cfec8d-8e15-4812-8a62-6036fe2d362c
- https://3v4l.org/EIKDT
- https://3v4l.org/4p0B8